File: tts/tts.cc
#include <iostream>
#include <cstring>
#include <cstdlib>
using namespace std;

int main(){
    // text변수에 원하는 text 집어넣으면 되용 함수로 만드실거면 함수 파라미터로 받아서 집어넣으셔도 되고!
    string text = "안녕하세요. 반갑습니다. 40m남았습니다. 야호!";
  
    // tts.php파일을 실행시켜 tts.mp3 파일을 만드는 코드
    string cmd = "php tts.php "+text;
    system(cmd.c_str());
   
    // mp3 실행 코드 추가 시켜주세요

    return 0;
}

File: tts/tts.php
<?php
  header('Content-Type: text/html; charset=utf-8');
  $client_id = "jrighen3yj";
  $client_secret = "RcQs9AQEsgfObrskj1sxz8k4wfin7MKeqAqaUd3t";
  $str = "";
  foreach($argv as $data){
        if($data != "tts.php"){
            $str = $str.$data."";
        }
  }
  $encText = urlencode("$str");
  $postvars = "speaker=mijin&speed=0&text=".$encText;
  $url = "https://naveropenapi.apigw.ntruss.com/voice/v1/tts";
  $is_post = true;
  $ch = curl_init();
  curl_setopt($ch, CURLOPT_URL, $url);
  curl_setopt($ch, CURLOPT_POST, $is_post);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
  curl_setopt($ch,CURLOPT_POSTFIELDS, $postvars);
  $headers = array();
  $headers[] = "X-NCP-APIGW-API-KEY-ID: ".$client_id;
  $headers[] = "X-NCP-APIGW-API-KEY: ".$client_secret;
  curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
  $response = curl_exec ($ch);
  $status_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
  echo "status_code:".$status_code."<br>";
  curl_close ($ch);
  if($status_code == 200) {
    //echo $response;
    $fp = fopen("tts.mp3", "w+");
    fwrite($fp, $response);
    fclose($fp);
    echo "$str";
	} else {
    echo "Error 내용:".$response;
  }
?>
